README.md

```diff
@@ -330,6 +331,65 @@ end
 
 5. Your app is now able to access the authenticated `ProductsController` from the `HomeController` using session tokens.
 
+### Marking shops as uninstalled on app/uninstalled webhook
+
+To ensure OAuth still works with session tokens, your app must update its shop records in the event a shop uninstalls your app. You can be notified of these events by subscribing to the `app/uninstall` [webhook](https://shopify.dev/docs/admin-api/rest/reference/events/webhook).
+
+#### Setup app/uninstall webhook
+1. Use the `add_webhook` generator from ShopifyApp to setup the `app/uninstalled` webhook for the app.
+
+```shell script
+rails g shopify_app:add_webhook -t app/uninstalled -a {your_app_url}/webhooks/app_uninstalled
+```
+
+This will add the `app/uninstalled` webhook to your config and create the `AppUninstalledJob` job class for you.  
+
+2. Update the `AppUninstalledJob` job class to mark the Shop record as uninstalled.
+
+```ruby
+class AppUninstalledJob < ActiveJob::Base
+  def perform(args)
+    shop = Shop.find_by(shopify_domain: args[:shop_domain])
+
+    mark_shop_as_uninstalled(shop)
+  end
+
+  private
+
+  def mark_shop_as_uninstalled(shop)
+    shop.destroy! if shop
+  end
+end
+```
+
+In this example, the app marks the shop as uninstalled by deleting the Shop record.
+
+3. Define a background job to ensure existing shops also have the uninstall webhook setup.  
+In this app, the `RegisterWebhooksForActiveShop` job is defined to iterate all shop records in the database and configure the webhooks.
+
+```ruby
+class RegisterWebhooksForActiveShops < ApplicationJob
+  queue_as :default
+
+  def perform
+    register_webhooks_for_active_shops
+  end
+
+  private
+
+  def register_webhooks_for_active_shops
+    Shop.find_each do |shop|
+      ShopifyApp::WebhooksManagerJob.perform_now(
+        shop_domain: shop.shopify_domain,
+        shop_token: shop.shopify_token,
+        webhooks: ShopifyApp.configuration.webhooks
```

Comment:
Are there any implications/considerations if `ShopifyApp.configuration.webhooks` contains webhooks other than uninstall? Specifically if any of the other topics in the array are already registered?

If yes, we should probably call that out and/or provide an alternative that targets only the new topics (for example maybe an inline array, assuming it is safe/acceptable practice)

```ruby
ShopifyApp::WebhooksManagerJob.perform_now(
  # ...
  webhooks: [
    {topic: 'app/uninstalled', address: "#{ENV['APP_URL']}/webhooks/app_uninstalled", format: 'json'},
  ]
)
```

Reply:
These webhook registrations [happen on every OAuth process](https://github.com/Shopify/shopify_app/blob/master/app/controllers/shopify_app/callback_controller.rb#L17). So my assumption is that it's idempotent. Going to confirm this with the webhooks team.